src/generate/register.rs

```diff
@@ -32,56 +33,26 @@ pub fn render(
     let rty = rsize.to_ty()?;
     let description = util::escape_brackets(util::respace(&register.description.clone().unwrap()).as_ref());
 
-    let unsafety = unsafety(register.write_constraint.as_ref(), rsize);
+    //let unsafety = unsafety(register.write_constraint.as_ref(), rsize);
 
     let mut mod_items = vec![];
-    let mut reg_impl_items = vec![];
     let mut r_impl_items = vec![];
     let mut w_impl_items = vec![];
 
     let can_read = [Access::ReadOnly, Access::ReadWriteOnce, Access::ReadWrite].contains(&access);
     let can_write = access != Access::ReadOnly;
 
-    if access == Access::ReadWrite || access == Access::ReadWriteOnce {
-        reg_impl_items.push(quote! {
-            ///Modifies the contents of the register
-            #[inline(always)]
-            pub fn modify<F>(&self, f: F)
-            where
-                for<'w> F: FnOnce(&R, &'w mut W) -> &'w mut W
-            {
-                let bits = self.register.get();
-                self.register.set(f(&R { bits }, &mut W { bits }).bits);
-            }
-        });
-    }
-
     if can_read {
-        reg_impl_items.push(quote! {
-            ///Reads the contents of the register
-            #[inline(always)]
-            pub fn read(&self) -> R {
-                R { bits: self.register.get() }
-            }
-        });
-
+        let desc = format!("Reader for register {}", register.name);
         mod_items.push(quote! {
-            ///Value read from the register
-            pub struct R {
-                bits: #rty,
-            }
-        });
-
-        r_impl_items.push(quote! {
-            ///Value of the register as raw bits
-            #[inline(always)]
-            pub fn bits(&self) -> #rty {
-                self.bits
-            }
+            #[doc = #desc]
+            pub type R = crate::R<#rty, super::#_name_pc>;
         });
     }
 
     if can_write {
+<<<<<<< HEAD
```

Comment:
Uhm, conflict markers?

Reply:
It happens.
Fixed.

Reply:
Sure, just makes me wonder how you actually tested this... 🤔

Reply:
I made merge after PR and tests.